Code/Source/RGLSystemComponent.cpp

```diff
@@ -126,36 +128,87 @@ namespace RGL
             return;
         }
 
-        AZStd::unique_ptr<EntityManager> entityManager;
-        if (entity.FindComponent<EMotionFX::Integration::ActorComponent>())
-        {
-            entityManager = AZStd::make_unique<ActorEntityManager>(entity.GetId());
-        }
-        else if (entity.FindComponent(AZ::Render::MeshComponentTypeId))
-        {
-            entityManager = AZStd::make_unique<MeshEntityManager>(entity.GetId());
-        }
-        else
+        if (m_activeLidarCount < 1U)
         {
+            m_unprocessedEntities.emplace(entity.GetId());
             return;
         }
 
-        [[maybe_unused]] bool inserted = m_entityManagers.emplace(entity.GetId(), AZStd::move(entityManager)).second;
-        AZ_Error(__func__, inserted, "Object with provided entityId already exists.");
+        ProcessEntity(entity);
     }
 
     void RGLSystemComponent::OnEntityContextDestroyEntity(const AZ::EntityId& id)
     {
+        m_unprocessedEntities.erase(id);
         m_entityManagers.erase(id);
     }
 
     void RGLSystemComponent::OnEntityContextReset()
     {
         m_entityManagers.clear();
+        m_unprocessedEntities.clear();
         m_modelLibrary.Clear();
         m_rglLidarSystem.Clear();
     }
 
+    void RGLSystemComponent::OnLidarCreated([[maybe_unused]] const ROS2::LidarId& lidarId)
+    {
+        ++m_activeLidarCount;
+
+        if (m_activeLidarCount > 1U)
+        {
+            return;
+        }
+
+        RGLNotificationBus::Broadcast(&RGLNotifications::OnLidarsExist);
+        for (auto entityIdIt = m_unprocessedEntities.begin(); entityIdIt != m_unprocessedEntities.end();)
+        {
+            AZ::Entity* entity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, *entityIdIt);
+            AZ_Assert(entity, "Failed to find entity with provided id!");
```

Comment:
I just wonder... will this not happen, when entity will be removed when RGL Lidar is not present, and then we spawn RGL Lidar. Just to be sure, this will be handled via OnEntityContextDestroyEntity in RGLSystemComponent, right?

Reply:
OnEntityContextDestroyEntity tries to erase from both unprocessed entities as well as entity managers so that both cases are handled.